### PR TITLE
[1LP][RFR] Orchestration template collection

### PR DESCRIPTION
**Pull request by sshveta**
Collection conversion of orchestration template

### CODE REVIEW

**Comment by sshveta** [comment]
All test passes for ec2-west .
RHOS-11 has some environmental issues and requests are not passed on it.

**Comment by dajoRH** [comment]
Lint report for commit 1bc931b8eb4795b9821ab8479438955f04e73bb2:

`cfme/services/catalogs/orchestration_template.py`:
- :bangbang: Line 189:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 242:60: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 243:60: **E127** *continuation line over-indented for visual indent*

`cfme/tests/services/test_orchestration_template.py`:
- :bangbang: Line 70:101: **E501** *line too long (112 > 100 characters)*
- :bangbang: Line 71:101: **E501** *line too long (118 > 100 characters)*
- :bangbang: Line 105:101: **E501** *line too long (105 > 100 characters)*
- :bangbang: Line 113:101: **E501** *line too long (112 > 100 characters)*
- :heavy_exclamation_mark: Line 120:1: **W391** *blank line at end of file*

`cfme/tests/services/test_provision_stack.py`:
- :bangbang: Line 99:101: **E501** *line too long (110 > 100 characters)*
- :heavy_exclamation_mark: Line 212:1: **W391** *blank line at end of file*

`setup.py`:
- :bangbang: Line 157:101: **E501** *line too long (116 > 100 characters)*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

File: setup.py [hunk 1]
```diff
@@ -153,7 +153,9 @@
             ('generic_object_definitions = '
                 'cfme.generic_objects.definition:GenericObjectDefinitionCollection'),
             ('generic_objects = '
-                'cfme.generic_objects.instance:GenericObjectInstanceCollection')
+                'cfme.generic_objects.instance:GenericObjectInstanceCollection'),
+            ('orchestration_template = '
```
Comment (quarckster) [inline setup.py]: **REQUIRED**
Please plural form in collections names.

File: cfme/services/catalogs/orchestration_template.py [hunk 2]
```diff
@@ -207,7 +187,38 @@ def exists(self):
             return False
 
 
-@navigator.register(OrchestrationTemplate, 'All')
+@attr.s
+class OrchestrationTemplateCollection(BaseCollection):
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
class OrchestrationTemplatesCollection(BaseCollection):

File: cfme/services/catalogs/orchestration_template.py [hunk 3]
```diff
@@ -125,38 +126,17 @@ def is_displayed(self):
...
-                   'template_type': temp_type,
-                   'content': content})
-        view.add_button.click()
-        view = self.create_view(DetailsTemplateView)
-        view.flash.assert_success_message('Orchestration Template '
-                                          '"{}" was saved'.format(self.template_name))
+    template_type = attr.ib()
+    template_name = attr.ib()
+    description = attr.ib()
+    draft = attr.ib()
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
description and draft fields are not mandatory.

File: cfme/services/catalogs/orchestration_template.py [hunk 4]
```diff
@@ -207,7 +187,38 @@ def exists(self):
...
 
-@navigator.register(OrchestrationTemplate, 'All')
+@attr.s
+class OrchestrationTemplateCollection(BaseCollection):
+    """A collection for the :py:class:`cfme.services.catalogs.orchestration_template`"""
+    ENTITY = OrchestrationTemplate
+
+    def instantiate(self, template_type, template_name, description, draft, content):
+        return self.ENTITY.from_collection(self, template_type, template_name,
+                                           description, draft, content)
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
`instantiate` is not needed here.

File: cfme/services/catalogs/orchestration_template.py [hunk 5]
```diff
@@ -207,7 +187,38 @@ def exists(self):
...
+
+    def create(self, template_name, description, template_type, draft=None, content=None):
+        self.template_type = template_type
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **QUESTION**
What's the purpose of this?

File: cfme/services/catalogs/orchestration_template.py [hunk 6]
```diff
@@ -207,7 +187,38 @@ def exists(self):
...
+        view = navigate_to(self, 'AddTemplate')
+        if template_type == "CloudFormation Templates":
+            temp_type = "Amazon CloudFormation"
+        elif template_type == "Heat Templates":
+            temp_type = "OpenStack Heat"
+        elif template_type == "Azure Templates":
+            temp_type = "Microsoft Azure"
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: *REQUIRED**
Please make a dictionary instead of `if elif`

Comment (sshveta) [reply cfme/services/catalogs/orchestration_template.py]: Pulling it from yamls now

File: cfme/services/catalogs/orchestration_template.py (hunk 2 above)
Comment (sshveta) [reply cfme/services/catalogs/orchestration_template.py]: Done

File: cfme/services/catalogs/orchestration_template.py (hunk 5 above)
Comment (sshveta) [reply cfme/services/catalogs/orchestration_template.py]: template type object is required here
https://github.com/ManageIQ/integration_tests/pull/6857/files#diff-b320dec2a69c9f5ff2599b8ef491dde5R248.

File: setup.py (hunk 1 above)
Comment (sshveta) [reply setup.py]: :+1:

File: cfme/services/catalogs/orchestration_template.py [hunk 7]
```diff
@@ -1,10 +1,11 @@
 # -*- coding: utf-8 -*-
+import attr
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
Please keep one blank line between external and internal imports.

File: cfme/services/catalogs/orchestration_template.py (hunk 3 above)
Comment (quarckster) [reply cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
I mean they should look like this:
```python
template_type = attr.ib()
template_name = attr.ib()
description = attr.ib(default=None)
draft = attr.ib(default=None)
```

File: cfme/services/catalogs/orchestration_template.py [hunk 8]
```diff
@@ -125,38 +126,16 @@ def is_displayed(self):
...
-                   'template_type': temp_type,
-                   'content': content})
-        view.add_button.click()
-        view = self.create_view(DetailsTemplateView)
-        view.flash.assert_success_message('Orchestration Template '
-                                          '"{}" was saved'.format(self.template_name))
+    template_type = attr.ib()
+    template_name = attr.ib()
+    description = attr.ib()
+    content = attr.ib()
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
I mean they should look like this:
```python
template_type = attr.ib()
template_name = attr.ib()
content = attr.ib()
description = attr.ib(default=None)
draft = attr.ib(default=None)
```

File: cfme/tests/services/test_provision_stack.py [hunk 9]
```diff
@@ -86,14 +86,16 @@ def dialog_name():
...
     template_type = provisioning['stack_provisioning']['template_type']
     template_name = fauxfactory.gen_alphanumeric()
-    template = OrchestrationTemplate(template_type=template_type,
-                                     template_name=template_name)
     file = provisioning['stack_provisioning']['data_file']
     data_file = load_data_file(str(orchestration_path.join(file)))
-    template.create(data_file.read().replace('CFMETemplateName', template_name))
+    content = data_file.read().replace('CFMETemplateName', template_name)
+
+    template = appliance.collections.orchestration_template.\
```
Comment (quarckster) [inline cfme/tests/services/test_provision_stack.py]: **REQUIRED**
```python
collection = appliance.collections.orchestration_templates
template = collection.create(template_type=template_type, template_name=template_name, description="my template", content=content)
```

File: cfme/tests/services/test_orchestration_template.py [hunk 10]
```diff
@@ -55,70 +54,74 @@ def create_template():
...
 
-def test_copy_template(provisioning, create_template):
+def test_copy_template(appliance, provisioning, create_template):
     template_type = provisioning['stack_provisioning']['template_type']
-    template = OrchestrationTemplate(template_type=template_type,
-                                     template_name=fauxfactory.gen_alphanumeric(),
-                                     description="my template")
-    template.create(create_template)
+    temp_type = provisioning['stack_provisioning']['template_type_dd']
+    template = appliance.collections.orchestration_templates.\
```
Comment (quarckster) [inline cfme/tests/services/test_orchestration_template.py]: **REQUIRED**
Please don't use `\` to break lines
```python
collection = appliance.collections.orchestration_templates
```

File: cfme/services/catalogs/orchestration_template.py [hunk 11]
```diff
@@ -207,7 +188,29 @@ def exists(self):
...
+    ENTITY = OrchestrationTemplate
+
+    def create(self, template_name, description, template_type,
+               temp_type, draft=None, content=None):
+        self.template_type = template_type
+        view = navigate_to(self, 'AddTemplate')
+        view.fill({'name': template_name,
+                   'description': description,
+                   'template_type': temp_type,
+                   'content': content})
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
```python
view.fill({'name': template_name,
           'description': description,
           'template_type': temp_type,
           'draft': draft,
           'template_type': temp_type,
           'content': content})
```

File: cfme/services/catalogs/orchestration_template.py [hunk 12]
```diff
@@ -207,7 +188,29 @@ def exists(self):
...
+        view.add_button.click()
+        template = self.instantiate(template_type=template_type, description=description,
+                                    template_name=template_name, content=content)
```
Comment (quarckster) [inline cfme/services/catalogs/orchestration_template.py]: **REQUIRED**
```python
template = self.instantiate(template_type=template_type, description=description, template_name=template_name, content=content, draft=draft)
```

**Comment by dajoRH** [comment]
Lint report for commit c017af2cf92e3614cb4adc944ce407be5adda6dc:

`cfme/services/catalogs/orchestration_template.py`:
- :bangbang: Line 207:82: **E241** *multiple spaces after ','*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Pull request merged**